File: semILaby/laby1/palindrom.c
=#include <stdio.h>
int main()
{
	int n;
	printf("Wczytaj liczbę: ");
	scanf("%d", &n);
	int pod;
	printf("Wczytaj podstawę liczby: ");
	scanf("%d", &pod);

	int czy = n;
	int palindrom = 0;
	while(n>0)
	{
		int q = n%pod;
		palindrom = palindrom*pod+q;
		n/=pod;
	}
	if(palindrom==czy)
	{
		printf("Ta liczba jest palindromem\n");
	}
	else
	{
		printf("Ta liczba nie jest palindromem\n");
	}
	return 0;
}

File: semILaby/laby1/palindrom.py
def main():
    n=int(input("Wczytaj liczbę: "))
    pod=int(input("Wczytaj jej podstawę: "))
    czy = n
    palindrom = 0
    while n>0 :
        q = n%pod
        palindrom = palindrom*pod+q
        n=n//pod
    if palindrom==czy :
        print(f"Ta liczba jest palindromem")
    else :
        print(f"Ta liczba nie jest palindromem")
if __name__=="__main__":
    main()
